### PR TITLE
add  prop to recive a reference to the underlying input element

### PR DESCRIPTION
**Pull request by janebitovi**
fix (REACT-107)

### CODE REVIEW

File: libs/react-numerics/src/lib/formatted-input.tsx [hunk 1]
```diff
@@ -63,4 +65,6 @@ export interface FormattedInputProps
   onChange: (value: string, changeType: ChangeType) => void;
   /** Pass a handler to be notified of key down events. */
   onKeyDown?: (evt: React.KeyboardEvent<HTMLInputElement>) => void;
+  /** Pass a reference back up to the underlying input element */
+  inputRef?: React.RefObject<HTMLInputElement>;
```
Comment (christopherjbaker) [inline libs/react-numerics/src/lib/formatted-input.tsx:69]: My only question would be about `RefObject` vs `MutableRefObject`. Should we accept both? Does one inherit from the other? I'm not fully aware of the differences, I just know in a couple specific places I've had to use one over the other.

**Review by christopherjbaker — APPROVED** [review]

File: libs/react-numerics/src/lib/formatted-input.tsx (hunk 1 above)
Comment (janebitovi) [reply libs/react-numerics/src/lib/formatted-input.tsx:69]: had a quick convo on the team call, going to leave as is with normal ref, but we can update it to allow mutable version if a use case comes up

Comment (christopherjbaker) [reply libs/react-numerics/src/lib/formatted-input.tsx:69]: perfect!

**Comment by rlmcneary2** [comment]
@janebitovi @christopherjbaker I didn't know anyone was even working on this! Thank you!

Per the issue I originally wrote (#11) I'd like the `ref` to be passed using React's standard mechanism of `forwardRef`. So for example `FormattedInput` would look like below.

```tsx
export const FormattedInput = forwardRef<Props, HTMLInputElement>((props, ref) => {
  return <input {...props} ref={ref} />
});
```

I'd like to see either this PR reopened or a new PR to make that change.

**Comment by christopherjbaker** [comment]
Using `forwardRef` is appropriate when the ref you are getting is to the component itself. When getting a ref to an underlying element, the convention is to use `somethingRef`. These are not mutually exclusive options, though. For example, [MUI's `Input`](https://mui.com/material-ui/api/input/) exposes a forwarded ref (which points to the root element, one of the wrapper divs) as well as an `inputRef` (which points to the input itself). If we were to provide a declarative interface for setting the value or otherwise hooking into the transform, that would be a perfect use case for `forwardRef` here, but I don't think we need that at the moment.

**Pull request merged**